include/LightGBM/c_api.h

```diff
@@ -258,69 +306,83 @@ DllExport int LGBM_BoosterUpdateOneIter(BoosterHandle handle, int* is_finished);
 * \param grad gradient statistics
 * \param hess second order gradient statistics
 * \param is_finished 1 means finised(cannot split any more)
-* \return 0 when success, -1 when failure happens
+* \return 0 when succeed, -1 when failure happens
 */
 DllExport int LGBM_BoosterUpdateOneIterCustom(BoosterHandle handle,
   const float* grad,
   const float* hess,
   int* is_finished);
 
 /*!
-* \brief get evaluation for training data and validation data
+* \brief Rollback one iteration
 * \param handle handle
-* \param data 0:training data, 1: 1st valid data, 2:2nd valid data ...
-* \param out_len len of output result
-* \param out_result the string containing evaluation statistics, should allocate memory before call this function
-* \return 0 when success, -1 when failure happens
+* \return 0 when succeed, -1 when failure happens
 */
-DllExport int LGBM_BoosterEval(BoosterHandle handle,
-  int data,
-  int64_t* out_len,
-  float* out_results);
+DllExport int LGBM_BoosterRollbackOneIter(BoosterHandle handle);
+
+/*!
+* \brief Get iteration of current boosting rounds
+* \return iteration of boosting rounds
+*/
+DllExport int LGBM_BoosterGetCurrentIteration(BoosterHandle handle, int64_t* out_iteration);
```

Comment:
In line with a previous suggestion that we avoid ambiguity by referring to the same concept with different terms ("num_..._models", "num_..._trees", "num_..._iterations") by just calling all of that 'trees', I would say this function would be more consistent as `LGBM_BoosterGetNumTrees()`. `out_iteration` would then be `out_num_trees`. The function description would then be "Get the current number of trees in the model".